src/orchestrator.ts

```diff
@@ -86,14 +86,51 @@ export class Orchestrator {
             let g = gen.next(undefined);
 
             while (true) {
+
+                if (!(g.value instanceof Task || g.value instanceof TaskSet)) {
+                    if (!g.done) {
+                        // The orchestrator must have yielded a non-Task related type,
+                        // so just return execution flow with what they yielded back.
+                        g = gen.next(g.value);
+                        continue;
+                    } else {
+                        log("Iterator is done");
```

Comment:
Where does this log statement go? Does it show up in the user logs?

Reply:
Not sure. I copied it from the other place where we mark the iterator as completed.